app/admin/documents.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+ActiveAdmin.register Document do
+  menu :priority => 12
+
+  permit_params :file, :event_type, :file_name, :url
+
+  controller do
+    def create
+      attrs = params[:document]
+      @document = Document.new
+      @document[:event_type] = attrs[:event_type]
```

Comment:
Lines 12-18 _could_ be factored into a method so it can be reused in the update method, but not a big deal, just something to consider for the future.